### PR TITLE
Ts-745

### PR DESCRIPTION
**Pull request by placccebo**

### CODE REVIEW

**Comment by placccebo** [comment]
[run tests]

**Comment by placccebo** [comment]
[run tests]

File: tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/util/multiNode/CommonCheckpointGroupServerTtlUtils.scala [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    })
+  }
+
+  def hasLogFiles(ledgerDirectory: File, numberOfFiles: Int): Boolean = {
+    val logs = ArrayBuffer[Integer]()
+    ledgerDirectory.listFiles().foreach(file => {
+      if (file.isFile) {
+        val name: String = file.getName
+        if (name.endsWith(".log")) {
+          logs += name.split("\\.")(0).toInt
```
Comment (tomskikh) [inline tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/util/multiNode/CommonCheckpointGroupServerTtlUtils.scala]: This array of integers isn't used. Just use counter or method ledgerDirectory.listFiles().count(...).

File: t-streams/src/test/scala/com/bwsw/tstreams/agents/producer/OpenTransactionsKeeperTests.scala [hunk 2]
```diff
@@ -21,7 +21,7 @@ package com.bwsw.tstreams.agents.producer
 
 
 import com.bwsw.tstreams.agents.group.ProducerTransactionState
-import com.bwsw.tstreams.testutils.IncreasingGenerator
+import com.bwsw.tstreams.IncreasingGenerator
```
Comment (bwsw) [inline t-streams/src/test/scala/com/bwsw/tstreams/agents/producer/OpenTransactionsKeeperTests.scala]: Increasing (What?) Generator?

File: tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/IdleCommonCheckpointGroupServerTtlTest.scala [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      ledgersExistInZkTree(trees, createdLedgers) shouldBe true
+
+      Thread.sleep(toMs(timeToWaitEntitiesDeletion) + gcWaitTimeMs)
+
+      val secondPartOfCreatedLedgers = timeToWaitEntitiesDeletion / maxIdleTimeBetweenRecords * treeFactor
+      ledgersExistInBookKeeper(fixture.ledgerManager, secondPartOfCreatedLedgers + createdLedgers) shouldBe false
+      ledgersExistInZkTree(trees, secondPartOfCreatedLedgers + createdLedgers) shouldBe false
+    })
+  }
+}
```
Comment (bwsw) [inline tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/IdleCommonCheckpointGroupServerTtlTest.scala]: Add a new line

File: tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/util/Utils.scala [hunk 4]
```diff
@@ -52,12 +53,26 @@ import scala.util.{Random, Try}
...
+  }
+
+  def deleteTempDirectories(): Unit = {
+    tmpDirs.foreach(dir => FileUtils.deleteDirectory(new File(dir)))
+    tmpDirs.clear()
+  }
+
+  def uuid: String = java.util.UUID.randomUUID.toString
 
   def startZkServerAndGetIt: (TestingServer, CuratorFramework) = {
```
Comment (bwsw) [inline tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/util/Utils.scala]: May be just rename to startZookeeperServer

File: tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/IdleCommonCheckpointGroupServerTtlTest.scala [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    ensembleNumber max writeQuorumNumber max ackQuorumNumber
+
+  case class FixtureParam(zkClient: CuratorFramework,
+                          ledgerManager: LedgerManager)
+
+  def withFixture(test: OneArgTest): Outcome = {
+    val (zkServer, zkClient, bookieServers) =
+      startZkAndBookieServerWithConfig(bookiesNumber, gcWaitTimeMs, entryLogSizeLimit)
+
+    val zk = ZooKeeperClient.newBuilder.connectString(zkClient.getZookeeperClient.getCurrentConnectionString).build
```
[the remark this answers is not in the document]
Comment (placccebo) [reply tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/IdleCommonCheckpointGroupServerTtlTest.scala]: yes, should be closed

File: tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/IdleCommonCheckpointGroupServerTtlTest.scala [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    val ledgerManagerFactory = LedgerManagerFactory.newLedgerManagerFactory(bookieServers.head._2, zk)
+    val ledgerManager = ledgerManagerFactory.newLedgerManager
+
+    val fixtureParam = FixtureParam(zkClient, ledgerManager)
+
+    Try {
+      withFixture(test.toNoArgTest(fixtureParam))
+    } match {
+      case Success(x) =>
+        ledgerManager.close()
```
[the remark this answers is not in the document]
Comment (placccebo) [reply tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/IdleCommonCheckpointGroupServerTtlTest.scala]: ok, thx

File: tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/util/Utils.scala (hunk 4 above)
Comment (placccebo) [reply tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/util/Utils.scala]: ok

**Comment by placccebo** [comment]
[run tests]

**Comment by placccebo** [comment]
[run tests]

**Comment by placccebo** [comment]
[run tests]

**Pull request merged**